sdks/python/apache_beam/runners/worker/sdk_worker.py

```diff
@@ -40,12 +40,21 @@
 class SdkHarness(object):
   REQUEST_METHOD_PREFIX = '_request_'
 
-  def __init__(self, control_address, worker_count):
+  def __init__(self, control_address, worker_count,
+               control_channel=None, credentials=None):
     self._worker_count = worker_count
     self._worker_index = 0
-    self._control_channel = grpc.intercept_channel(
-        grpc.insecure_channel(control_address), WorkerIdInterceptor())
-    self._data_channel_factory = data_plane.GrpcClientDataChannelFactory()
+    if control_channel is None:
+      logging.info('Creating insecure channel.')
+      self._control_channel = grpc.intercept_channel(
+          grpc.insecure_channel(control_address), WorkerIdInterceptor())
+      self._data_channel_factory = data_plane.GrpcClientDataChannelFactory()
+    else:
+      logging.info('Using provided channel.')
+      self._control_channel = grpc.intercept_channel(
+          control_channel, WorkerIdInterceptor())
+      self._data_channel_factory = data_plane.GrpcClientDataChannelFactory(
+          credentials)
```

Comment:
Safe to just pass credentials in both cases, right?

Reply:
done.